lib/lte_link_control/lte_lc.c

```diff
@@ -566,11 +552,50 @@ static int enable_notifications(void)
 	return 0;
 }
 
+static int fallback_system_mode_set(void)
+{
+	int err;
+	enum lte_lc_system_mode fallback_sys_mode;
+
+	switch (lte_lc_sys_mode) {
+	case LTE_LC_SYSTEM_MODE_LTEM:
+		fallback_sys_mode = LTE_LC_SYSTEM_MODE_NBIOT;
+		break;
+
+	case LTE_LC_SYSTEM_MODE_NBIOT:
+		fallback_sys_mode = LTE_LC_SYSTEM_MODE_LTEM;
+		break;
+
+	case LTE_LC_SYSTEM_MODE_LTEM_GPS:
+		fallback_sys_mode = LTE_LC_SYSTEM_MODE_NBIOT_GPS;
+		break;
+
+	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
+		fallback_sys_mode = LTE_LC_SYSTEM_MODE_LTEM_GPS;
+		break;
+
+	default:
+		LOG_DBG("Fallback enabled, but not possible from current system mode");
```

Comment:
I made this a debug log, because this is not an error. Fallback is possible from some system modes, but not from all of them. I would keep the level as it is now. Not sure if there's need to log the current system mode, but I can add it if you want. This PR already changes `lte_lc_system_mode_set()` so, that system mode changes are logged.

Reply:
Isn't this log something we'd like to be aware of even if debug logging is not turned ON? Developer might have misconfigured the application or used system modes in an undesired manner. Whether it would be LOG_WRN or maybe LOG_INF is another question. I'm not sure LOG_WRN is wrong or a spamming too much here.

Logging system mode adds value as this log is not normal and having the necessary information for the "problem" in the log is nice. And doesn't really cost anything.

Reply:
Currently, `CONFIG_LTE_NETWORK_USE_FALLBACK` is enabled by default, unless network mode is LTE-M + NB-IoT or LTE-M + NB-IoT + GNSS. Thus, it's enabled by default when `CONFIG_LTE_NETWORK_MODE_DEFAULT` is selected, but fallback can not be performed. If we make this LOG_WRN, I think we should change the Kconfig so, that fallback is not enabled with `CONFIG_LTE_NETWORK_MODE_DEFAULT` either.

Reply:
Updated as agreed offline.